compiler/rustc_errors/src/lib.rs

```diff
@@ -366,12 +367,25 @@ pub struct ExplicitBug;
 
 impl fmt::Display for ExplicitBug {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "parser internal bug")
+        write!(f, "internal bug")
     }
 }
 
 impl error::Error for ExplicitBug {}
 
+/// Signifies that the compiler died with an explicit call to `.delay_good_path_bug`
+/// rather than a failed assertion, etc.
+#[derive(Copy, Clone, Debug)]
+pub struct GoodPathBug;
+
+impl fmt::Display for GoodPathBug {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "good path bug")
+    }
+}
```

Comment:
is this impl (or the `Error` impl) ever used?

Reply:
It's conventional for types implementing the `Error` trait:

> Errors must describe themselves through the [Display](https://doc.rust-lang.org/std/fmt/trait.Display.html) and [Debug](https://doc.rust-lang.org/std/fmt/trait.Debug.html) traits. Error messages are typically concise lowercase sentences without trailing punctuation [...]

Reply:
Right, but I'm asking whether this needs to implement `Error` in the first place. I don't see that bound ever used.

Reply:
Oh, didn't see the parenthetical. Uh, I'll see.